src/coreclr/gc/gcpriv.h

```diff
@@ -4391,6 +4485,10 @@ class gc_heap
     PER_HEAP_ISOLATED_FIELD_DIAG_ONLY BOOL heap_analyze_enabled;
 #endif //HEAP_ANALYZE
 
+#ifdef MULTIPLE_HEAPS
+    PER_HEAP_FIELD bool uoh_msl_before_gc_p;
```

Comment:
this should be under the `#ifdef STRESS_DYNAMIC_HEAP_COUNT` check shoiuldn't it?